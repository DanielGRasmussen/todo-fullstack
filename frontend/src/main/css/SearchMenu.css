/* Menu */
#menu {
	display: grid;
	grid-template-columns: 1fr 1fr;
	grid-template-rows: 30px 60px 60px 30px 60px;
	grid-gap: 15px;
	margin-bottom: 25px;
}

#filters-label {
	grid-column: 1 / 3;
	grid-row: 1 / 2;
	margin-top: 16px;
	text-align: center;
	cursor: default;
}

#sorts-label {
	grid-column: 1 / 3;
	grid-row: 4 / 5;
	margin-top: 16px;
	text-align: center;
	cursor: default;
}

/* Filters */
#status {
	grid-column: 1 / 2;
	grid-row: 2 / 3;
	margin-left: auto;
}

#type {
	grid-column: 1 / 2;
	grid-row: 3 / 4;
	margin-left: auto;
}

/* Datepicker */
#datepicker-wrapper {
	width: 372px;
	grid-column: 2 / 3;
	grid-row: 2 / 3;
}

#datepicker {
	width: 359px;
	height: 46px;
	font-size: 12pt;
	border-radius: 5px;
	border: 1px solid #9e9e9e;
	padding-left: 7px;

	grid-column: 2 / 3;
	display: block;
}

#datepicker:focus {
	border: 1px solid #2684ff;
}

#time {
	grid-column: 2 / 3;
	grid-row: 3 / 4;
}

/* Selects sorts */
#menu #sorts {
	display: flex;
	grid-column: 1 / 3;
	grid-row: 5 / 6;
	margin: 0 auto;
}

/* Search bar */
#search-bar {
	width: 250px;
	height: 46px;
	font-size: 12pt;
	border: 1px solid #9e9e9e;
	border-radius: 4px;
	padding-left: 8px;
}

@media screen and (min-width: 801px) {
	#search-bar-container-container {
		width: 100vw;
		position: fixed;
		top: -100px;
		left: 0;
		z-index: 1;
		transition: top 0.3s ease-in-out;
		height: 70px;
	}

	#search-bar-container {
		width: 80%;
		position: relative;
		top: 0;
		min-width: 800px;
		max-width: 1100px;
		margin: 0 auto;
	}

	#menu #search-bar {
		position: absolute;
		top: 117px;
		right: 15px;
		height: 40px;
		border-radius: 15px;
	}

	#menu #search-bar-container-container.hide {
		/* This is -200px instead of -100px so the animation speed stays the same and the containers don't block the nav */
		top: -200px;
	}
}

/* Tablet/mobile SearchMenu */
@media screen and (max-width: 800px) {
	#menu {
		display: block;
	}

	#menu label {
		margin-top: 10px;
		display: block;
	}

	#search-bar {
		width: 80%;
		margin: 0 auto;
		display: block;
	}

	#menu > div {
		margin: 0 auto;
		width: 80%;
		min-width: 300px;
	}

	#datepicker-wrapper {
		width: 100%;
	}

	#datepicker {
		width: calc(100% - 11px);
	}

	.css-b62m3t-container .css-u5xlfi-control,
	.css-b62m3t-container .css-10v2721-control {
		width: 100%;
		margin: 0 auto;
	}

	#sort-select {
		width: calc(100% - 50px);
	}

	.dropdown-item-header {
		cursor: pointer;
		position: relative;
	}

	.arrow {
		position: absolute;
		top: 13px;
		right: 20px;
		width: 0;
		height: 0;
		margin-left: 5px;
		border-style: solid;
		border-width: 5px 4px 0 4px;
		border-color: #4e4e4e transparent transparent transparent;
		transition: transform 0.3s ease-in-out;
	}

	.arrow.up {
		transform: rotate(180deg);
	}

	.dropdown-item-content {
		transition: all 0.3s ease-in-out;
		height: 0;
		overflow: hidden;
	}

	/* fit-content doesn't show the animation for whatever reason */
	.dropdown-item-content.filters.show {
		height: 324px;
	}

	.dropdown-item-content.sorts.show {
		height: 50px;
	}
}

/* React-select animations */
.select-menu {
	animation: fadeIn 0.2s ease-in-out;
}

.select-menu-close {
	animation: fadeOut 0.3s ease-in-out;
}

@keyframes fadeIn {
	from {
		opacity: 0;
		transform: translateY(2rem);
	}
	to {
		opacity: 1;
		transform: translateY(0);
	}
}

@keyframes fadeOut {
	from {
		opacity: 1;
		transform: translateY(0);
	}
	to {
		opacity: 0;
		transform: translateY(2rem);
	}
}

/* Style for the tooltips */
.tooltip {
	position: relative;
}

.tooltip::before {
	display: none;
	position: absolute;
	background-color: black;
	color: white;
	padding: 5px;
	border-radius: 5px;
	top: -30px;
	left: 50%;
	transform: translateX(-50%);
	text-align: center;
}

/* Show the tooltip on hover */
.tooltip:hover::before {
	display: block;
}

/* Sort-direction title message in SortButton.css */
