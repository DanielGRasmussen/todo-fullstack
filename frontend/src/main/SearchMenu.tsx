import "./css/SearchMenu.css";
import React, { useEffect, useState } from "react";
import makeAnimated from "react-select/animated";
import Select, { components } from "react-select";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";
import SortButton from "./SortButton";

const animatedComponents = makeAnimated();
const customComponents = {
	Menu: (props) => (
		<components.Menu children="" {...props} className="select-menu" />
	)
};

function select_on_close(uniqueId: string): void {
	// Can't just do event handlers with class toggle and timeout
	// Menu reappears for a split second and animation goes ultra-fast if you do.
	const menu: Element = document.querySelector(`#${uniqueId} .select-menu`);
	const container: HTMLElement = menu?.parentElement;
	const clonedMenu: HTMLElement = menu?.cloneNode(true) as HTMLElement;

	if (!clonedMenu) return;

	clonedMenu.classList.add("select-menu-close");
	clonedMenu.addEventListener("animationend", () => {
		container?.removeChild(clonedMenu);
	});

	container?.appendChild(clonedMenu);
}

interface ISearchMenu {
	searchQuery: string;
	setSearchQuery;
	sortingOptions: { value: string; label: string }[];
	selectedSortingOption: { value: string; label: string }[];
	setSelectedSortingOption;
	sortOrder: boolean;
	setSortOrder;
	currentStatusFilters: { value: string; label: string }[];
	setStatusFilter;
	filterOptions: { value: string; label: string }[];
	filters: string[];
	setFilters;
	currentTimeframe: { value: string; label: string };
	setCurrentTimeframe;
	startDate;
	setStartDate;
	endDate;
	setEndDate;
}

function SearchMenu({
	searchQuery,
	setSearchQuery,
	sortingOptions,
	selectedSortingOption,
	setSelectedSortingOption,
	sortOrder,
	setSortOrder,
	currentStatusFilters,
	setStatusFilter,
	filterOptions,
	filters,
	setFilters,
	currentTimeframe,
	setCurrentTimeframe,
	startDate,
	setStartDate,
	endDate,
	setEndDate
}: ISearchMenu): JSX.Element {
	/* This renders the search menu with options to filter for the TodoList
	 *
	 * This component takes twelve props:
	 * searchQuery: The current search query entered by the user.
	 * setSearchQuery: A function to update the search query based on user input.
	 * sortingOptions: An array of sorting options to be displayed in the sorting dropdown menu.
	 * selectedSortingOption: The currently selected sorting option.
	 * setSelectedSortingOption: A function to update the selected sorting option based on user selection.
	 * sortOrder: The current sorting order (ascending or descending).
	 * setSortOrder: A function to update the sorting order based on user selection.
	 * currentStatusFilters: The currently selected status filters.
	 * setStatusFilter: A function to update the selected status filters based on user selection.
	 * filterOptions: An array of filter options to be displayed in the type filter dropdown menu.
	 * filters: The currently selected type filters.
	 * setFilters: A function to update the selected type filters based on user selection.
	 * currentTimeframe: The currently selected timeframe.
	 * setCurrentTimeframe: A function to update the selected type timeframe based on user selection.
	 *
	 * The component returns a form with several search inputs such as a search bar, a select dropdown for status
	 * filters, a select dropdown for type filters, and a select dropdown for sorting options. The form also includes a
	 * button to toggle the sort order.
	 */
	const [filterOpen, setfilterOpen] = useState(false);
	const [sortOpen, setsortOpen] = useState(false);
	const [windowWidth, setWindowWidth] = useState(window.innerWidth);

	function toggleFilters() {
		setfilterOpen(!filterOpen);
	}

	function toggleSorts() {
		setsortOpen(!sortOpen);
	}

	useEffect(() => {
		function handleResize() {
			setWindowWidth(window.innerWidth);
		}

		window.addEventListener("resize", handleResize);
		return () => {
			window.removeEventListener("resize", handleResize);
		};
	});

	// Makes react-select thinner
	const selectStyles = {
		control: (provided) => ({
			...provided,
			background: "#fff",
			borderColor: "#9e9e9e",
			minHeight: "50px",
			height: "50px",
			boxShadow: null,
			width: "370px",
			cursor: "pointer",
			textTransform: "capitalize"
		}),
		valueContainer: (provided) => ({
			...provided,
			height: "30px",
			padding: "0 6px"
		}),
		indicatorSeparator: () => ({
			display: "none"
		}),
		indicatorsContainer: (provided) => ({
			...provided,
			height: "40px",
			marginTop: "5px"
		}),
		menu: (provided) => ({
			...provided,
			marginTop: "-5px",
			width: "370px",
			textTransform: "capitalize"
		})
	};

	function dateSelectionChange(dates) {
		const [start, end] = dates;
		setStartDate(start);
		setEndDate(end);
	}

	const statusOptions = [
		{ value: "incomplete", label: "Incomplete" },
		{ value: "in-progress", label: "In-Progress" },
		{ value: "complete", label: "Complete" }
	];

	const timeframeOptions = [
		{ value: "behind", label: "Behind" },
		{ value: "today", label: "Today" },
		{ value: "tomorrow", label: "Tomorrow" },
		{ value: "week", label: "This week" },
		{ value: "month", label: "This month" },
		{ value: "year", label: "This year" },
		{ value: "all", label: "All" }
	];

	const filterHeader = <h2 id="filters-label">Filters</h2>;
	const filterElements = (
		<>
			<div id="status">
				<label htmlFor="status-filter">Status filter:</label>
				<Select
					id="status-filter"
					options={statusOptions}
					value={currentStatusFilters}
					onChange={(options) => setStatusFilter(options)}
					styles={selectStyles}
					isMulti
					components={{ ...animatedComponents, ...customComponents }}
					onMenuClose={() => {
						select_on_close("status-filter");
					}}
				/>
			</div>
			<div id="type">
				<label htmlFor="type-filter">Type filter:</label>
				<Select
					id="type-filter"
					options={filterOptions}
					value={filters}
					onChange={(options) => setFilters(options)}
					styles={selectStyles}
					isMulti
					components={{ ...animatedComponents, ...customComponents }}
					onMenuClose={() => {
						select_on_close("filter-select");
					}}
				/>
			</div>
			<div id="datepicker-wrapper">
				{/* For the clearing element. */}
				<label htmlFor="datepicker">Date range:</label>
				<DatePicker
					selected={startDate}
					onChange={dateSelectionChange}
					startDate={startDate}
					endDate={endDate}
					selectsRange
					isClearable
					placeholderText="Select date range"
					id="datepicker"
				/>
			</div>
			<div id="time">
				<label htmlFor="timeframe">Timeframe:</label>
				<Select
					id="timeframe"
					options={timeframeOptions}
					value={currentTimeframe}
					onChange={(option) => setCurrentTimeframe(option)}
					styles={selectStyles}
					isSearchable={false}
					components={{ ...animatedComponents, ...customComponents }}
					onMenuClose={() => {
						select_on_close("timeframe");
					}}
				/>
			</div>
		</>
	);

	const filtersDropdown = (
		<div className="dropdown-item">
			<div className="dropdown-item-header" onClick={toggleFilters}>
				{filterHeader}
				<span className={`arrow ${filterOpen ? "up" : "down"}`} />
			</div>

			<div
				className={`dropdown-item-content filters ${
					filterOpen ? "show" : null
				}`}
			>
				{filterElements}
			</div>
		</div>
	);

	const sortHeader = <h2 id="sorts-label">Sorts</h2>;
	const sortElements = (
		<div id="sorts">
			<Select
				id="sort-select"
				options={sortingOptions}
				value={selectedSortingOption}
				onChange={(options) => setSelectedSortingOption(options)}
				styles={selectStyles}
				isMulti
				components={{ ...animatedComponents, ...customComponents }}
				onMenuClose={() => {
					select_on_close("sort-select");
				}}
				placeholder="Sort options"
			/>
			<SortButton sortOrder={sortOrder} setSortOrder={setSortOrder} />
		</div>
	);

	const sortDropdown = (
		<div className="dropdown-item">
			<div className="dropdown-item-header" onClick={toggleSorts}>
				{sortHeader}
				<span className={`arrow ${sortOpen ? "up" : "down"}`} />
			</div>

			<div
				className={`dropdown-item-content sorts ${
					sortOpen ? "show" : null
				}`}
			>
				{sortElements}
			</div>
		</div>
	);

	return (
		<form id="menu" autoComplete="off">
			<div id="search-bar-container-container">
				{/* First container for absolute positioning and 100% width */}
				<section id="search-bar-container">
					{/*
					 * Second container to be the same width as the header/main then use margins to move it. (Fancy CSS could probably replace that)
					 * Section instead of div because my IDE and ESLint didn't agree on how it should be formatted.
					 * This was not passed into the header function because the functions this calls on didn't to update the screen.
					 */}
					<input
						id="search-bar"
						type="text"
						value={searchQuery}
						onChange={(event) => setSearchQuery(event.target.value)}
						placeholder="Search..."
					/>
				</section>
			</div>
			{windowWidth > 800 ? filterHeader : null}
			{windowWidth < 800 ? filtersDropdown : filterElements}
			{windowWidth > 800 ? sortHeader : null}
			{windowWidth < 800 ? sortDropdown : sortElements}
		</form>
	);
}

export default SearchMenu;
