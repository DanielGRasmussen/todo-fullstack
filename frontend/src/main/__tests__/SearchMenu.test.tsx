import React from "react";
import { fireEvent, render } from "@testing-library/react";
import SearchMenu from "../SearchMenu";

describe("SearchMenu", () => {
	const setSearchQuery = jest.fn();
	const setSelectedSortingOption = jest.fn();
	const setStatusFilter = jest.fn();
	const setSortOrder = jest.fn();
	const setFilters = jest.fn();

	const props = {
		searchQuery: "",
		setSearchQuery,
		sortingOptions: [
			{ value: "option1", label: "Option 1" },
			{ value: "option2", label: "Option 2" }
		],
		selectedSortingOption: [{ value: "option1", label: "Option 1" }],
		setSelectedSortingOption,
		sortOrder: true,
		setSortOrder,
		currentStatusFilters: [{ value: "incomplete", label: "Incomplete" }],
		setStatusFilter,
		filterOptions: [
			{ value: "filter1", label: "Filter 1" },
			{ value: "filter2", label: "Filter 2" }
		],
		filters: [],
		setFilters
	};

	it("should render the component", () => {
		const { container } = render(<SearchMenu {...props} />);
		expect(container).toMatchSnapshot();
	});

	it("should call the setSearchQuery function", () => {
		render(<SearchMenu {...props} />);
		const searchBar = document.getElementById(
			"search-bar"
		) as HTMLInputElement;
		fireEvent.change(searchBar, { target: { value: "new value" } });
		expect(setSearchQuery).toHaveBeenCalledWith("new value");
	});
});
