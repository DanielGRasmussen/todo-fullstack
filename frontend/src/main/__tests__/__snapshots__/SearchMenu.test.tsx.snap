// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchMenu should render the component 1`] = `
<div>
  <form
    autocomplete="off"
    id="menu"
  >
    <div
      id="search-bar-container-container"
    >
      <section
        id="search-bar-container"
      >
        <input
          id="search-bar"
          placeholder="Search..."
          type="text"
          value=""
        />
      </section>
    </div>
    <h2
      id="filters-label"
    >
      Filters
    </h2>
    <div
      id="status"
    >
      <label
        for="status-filter"
      >
        Status filter:
      </label>
      <div
        class=" css-b62m3t-container"
        id="status-filter"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-u5xlfi-control"
        >
          <div
            class=" css-525zur-ValueContainer"
            style="display: flex;"
          >
            <div
              style="overflow: hidden; white-space: nowrap; width: auto;"
            >
              <div
                class=" css-1p3m7a8-multiValue"
              >
                <div
                  class=" css-wsp0cs-MultiValueGeneric"
                >
                  Incomplete
                </div>
                <div
                  aria-label="Remove Incomplete"
                  class=" css-12a83d4-MultiValueRemove"
                  role="button"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="14"
                    viewBox="0 0 20 20"
                    width="14"
                  >
                    <path
                      d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class=" css-qbdosj-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-2-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-1gscolb-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              class=" css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                />
              </svg>
            </div>
            <span
              class=" css-1xjgjl1-IndicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      id="type"
    >
      <label
        for="type-filter"
      >
        Type filter:
      </label>
      <div
        class=" css-b62m3t-container"
        id="type-filter"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-u5xlfi-control"
        >
          <div
            class=" css-qsn7pb-ValueContainer"
            style="display: grid;"
          >
            <div
              class=" css-1jqq78o-placeholder"
              id="react-select-3-placeholder"
            >
              Select...
            </div>
            <div
              class=" css-qbdosj-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-3-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-3-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-1gscolb-IndicatorsContainer"
          >
            <span
              class=" css-1xjgjl1-IndicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      id="datepicker-wrapper"
    >
      <label
        for="datepicker"
      >
        Date range:
      </label>
      <div
        class="react-date-picker react-date-picker--closed react-date-picker--enabled"
        id="search-datepicker"
      >
        <div
          class="react-date-picker__wrapper"
        >
          <div
            class="react-date-picker__inputGroup"
          >
            <input
              hidden=""
              max="275760-09-12"
              min="0001-01-01"
              name="date"
              style="visibility: hidden; position: absolute; z-index: -999;"
              type="date"
              value=""
            />
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
              data-input="true"
              inputmode="numeric"
              max="12"
              min="1"
              name="month"
              placeholder="--"
              type="number"
              value=""
            />
            <span
              class="react-date-picker__inputGroup__divider"
            >
              /
            </span>
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
              data-input="true"
              inputmode="numeric"
              max="31"
              min="1"
              name="day"
              placeholder="--"
              type="number"
              value=""
            />
            <span
              class="react-date-picker__inputGroup__divider"
            >
              /
            </span>
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
              data-input="true"
              inputmode="numeric"
              max="275760"
              min="1"
              name="year"
              placeholder="----"
              step="1"
              type="number"
              value=""
            />
          </div>
          <button
            class="react-date-picker__clear-button react-date-picker__button"
            type="button"
          >
            <svg
              class="react-date-picker__clear-button__icon react-date-picker__button__icon"
              height="19"
              stroke="black"
              stroke-width="2"
              viewBox="0 0 19 19"
              width="19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="4"
                x2="15"
                y1="4"
                y2="15"
              />
              <line
                x1="15"
                x2="4"
                y1="4"
                y2="15"
              />
            </svg>
          </button>
          <button
            class="react-date-picker__calendar-button react-date-picker__button"
            type="button"
          >
            <svg
              class="react-date-picker__calendar-button__icon react-date-picker__button__icon"
              height="19"
              stroke="black"
              stroke-width="2"
              viewBox="0 0 19 19"
              width="19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="15"
                width="15"
                x="2"
                y="2"
              />
              <line
                x1="6"
                x2="6"
                y1="0"
                y2="4"
              />
              <line
                x1="13"
                x2="13"
                y1="0"
                y2="4"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      id="time"
    >
      <label
        for="timeframe"
      >
        Timeframe:
      </label>
      <div
        class=" css-b62m3t-container"
        id="timeframe"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-4-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-u5xlfi-control"
        >
          <div
            class=" css-qsn7pb-ValueContainer"
          >
            <div
              class=" css-1dimb5e-singleValue"
              style="opacity: 1; transition: opacity 1ms;"
            >
              Today
            </div>
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-haspopup="true"
              aria-readonly="true"
              class="css-mohuvp-dummyInput-DummyInput"
              id="react-select-4-input"
              inputmode="none"
              role="combobox"
              tabindex="0"
              value=""
            />
          </div>
          <div
            class=" css-1gscolb-IndicatorsContainer"
          >
            <span
              class=" css-1xjgjl1-IndicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <h2
      id="sorts-label"
    >
      Sorts
    </h2>
    <div
      id="sorts"
    >
      <div
        class=" css-b62m3t-container"
        id="sort-select"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-5-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-u5xlfi-control"
        >
          <div
            class=" css-525zur-ValueContainer"
            style="display: flex;"
          >
            <div
              style="overflow: hidden; white-space: nowrap; width: auto;"
            >
              <div
                class=" css-1p3m7a8-multiValue"
              >
                <div
                  class=" css-wsp0cs-MultiValueGeneric"
                >
                  Option 1
                </div>
                <div
                  aria-label="Remove Option 1"
                  class=" css-12a83d4-MultiValueRemove"
                  role="button"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="14"
                    viewBox="0 0 20 20"
                    width="14"
                  >
                    <path
                      d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class=" css-qbdosj-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-5-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-1gscolb-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              class=" css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                />
              </svg>
            </div>
            <span
              class=" css-1xjgjl1-IndicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ascending tooltip"
        id="sort-direction"
      >
        <svg
          enable-background="new 0 0 50 150"
          viewBox="0 0 50 150"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <line
              stroke="#000"
              stroke-miterlimit="10"
              stroke-width="6"
              x1="23.871"
              x2="23.871"
              y1="149"
              y2="24.964"
            />
            <polygon
              points="42.292 25.87 3.969 25.87 24.885 1.917"
            />
          </g>
        </svg>
      </div>
    </div>
  </form>
</div>
`;
